### DISCUSSION TITLE
Can we set a custom endpoint?

### DISCUSSION THREAD

**Discussion by PsiACE** [3ru/gpt-translate#53] [Help] [answered]
### Summary

Due to the strict limitations on API and account by OpenAI, I have to try Azure OAI or other services that claim to provide compatible APIs. If it is possible to set a custom endpoint, at least some attempts can be made, although there may still be some compatibility issues.

**Comment by 3ru** [comment]
@PsiACE 
I'm not entirely sure I fully understand your current situation and what you're looking for, so let me clarify again:

Are you deploying an LLM on Azure and wish to use that model for translations? In other words, does this mean you want to use your own endpoint instead of the OpenAI endpoint (i.e., https://api.openai.com/v1/models)?

Or are you looking to use a completely different AI service (such as [Claude](https://www.anthropic.com/claude))?

I'm considering trying out Azure OAI and Claude 3 since there have been requests for them before, but am I correctly capturing what you're looking for?

**Comment by PsiACE** [comment]
I want to use my own endpoint (azure oai).😘

**Comment by 3ru** [comment, marked as answer]
@PsiACE 

I'm not sure if it works correctly, but can you try giving it as an argument `basePath` to action?


e.g
```yml
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          apikey: ${{ secrets.OPENAI_API_KEY }}
          model: "gpt-4-turbo-preview"
          basePath: "" # your endpoint here
```

---

The URL will probably consist of the following form
```js
// The name of your Azure OpenAI Resource.
// https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource?pivots=web-portal#create-a-resource
const resource = '<your resource name>';

// Corresponds to your Model deployment within your OpenAI resource, e.g. my-gpt35-16k-deployment
// Navigate to the Azure OpenAI Studio to deploy a model.
const model = '<your model>';

// your url
// https://${resource}.openai.azure.com/openai/deployments/${model}
```
ref: https://github.com/openai/openai-node/blob/31ddda89195849b09f80d920abba74a225535248/examples/azure.ts#L5-L21
